### PR TITLE
replace detector

### PR DESCRIPTION
**Pull request by matemax**

### CODE REVIEW

File: lunavl/sdk/base.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    """
+    Base class for estimation structures.
+
+    Attributes:
+        _coreEstimation: core estimation
+    """
+
+    __slots__ = ("_coreEstimation",)
+
+    def __init__(self, coreEstimation):
```
Comment (gluhar2006) [inline lunavl/sdk/base.py]: typing

File: lunavl/sdk/base.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    def coreEstimation(self):
+        """
+        Get core estimation from init
+        Returns:
+            _coreEstimation
+        """
+        return self._coreEstimation
+
+    @abstractmethod
+    def asDict(self) -> Union[dict, list]:
```
Comment (gluhar2006) [inline lunavl/sdk/base.py:32]: ?

Comment (matemax) [reply lunavl/sdk/base.py:32]: list for Landmarks

File: lunavl/sdk/base.py (hunk 1 above)
Comment (matemax) [reply lunavl/sdk/base.py]: done

File: lunavl/sdk/base.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        Args:
+            coreLandmarks (LANDMARKS): core landmarks
+        """
+        super().__init__(coreLandmarks)
+        self._points: Optional[Tuple[Point[float], ...]] = None
+
+    @property
+    def points(self) -> Tuple[Point[float], ...]:
+        """
+        Lazy load of points.
```
Comment (Zeskbest) [inline lunavl/sdk/base.py]:
```suggestion
        Lazy points loader.
```

File: lunavl/sdk/base.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        """
+        return str(self.asDict())
+
+
+class Landmarks(BaseEstimation):
+    """
+    Base class for landmarks
+
+    Attributes:
+        _points (Optional[Tuple[Point[float]]]): lazy load attributes, converted to point list core landmarks
```
Comment (Zeskbest) [inline lunavl/sdk/base.py]:
```suggestion
        _points (Optional[Tuple[Point[float]]]): lazy loaded attributes: core landmarks as point list
```

File: lunavl/sdk/base.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    def points(self) -> Tuple[Point[float], ...]:
+        """
+        Lazy load of points.
+
+        Returns:
+            list of points
+        """
+        if self._points is None:
+            self._points = tuple(
+                (Point.fromVector2(self._coreEstimation[index]) for index in range(len(self._coreEstimation)))
```
Comment (Zeskbest) [inline lunavl/sdk/base.py]:
```suggestion
                Point.fromVector2(self._coreEstimation[index]) for index in range(len(self._coreEstimation))
```

File: lunavl/sdk/base.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+class BoundingBox(BaseEstimation):
+    """
+    Detection bounding box, it is characterized of rect and score:
+
+        - rect (Rect[float]): face bounding box
+        - score (float): face score (0,1), detection score is the measure of classification confidence
+                         and not the source image quality. It may be used topick the most "*confident*" face of many.
+    """
+
+    #  pylint: disable=W0235
```
Comment (Zeskbest) [inline lunavl/sdk/base.py:106]: W0235 does not exist
http://pylint-messages.wikidot.com/all-codes

File: lunavl/sdk/detectors/base.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        return self._image
+
+    def asDict(self) -> Dict[str, Any]:
+        """
+        Convert face detection to dict (json).
+
+        Returns:
+            dict. required keys: 'rect', 'score'.
+        """
+        return {"rect": self.boundingBox.rect.asDict(), "score": self.boundingBox.score}
```
Comment (Zeskbest) [inline lunavl/sdk/detectors/base.py]:
```suggestion
        return self.boundingBox.asDict()
```

File: lunavl/sdk/errors/exceptions.py [hunk 8]
```diff
@@ -4,7 +4,9 @@
 from functools import wraps
 from typing import Optional, Any, Callable
 
-from lunavl.sdk.errors.errors import ErrorInfo
+from FaceEngine import FSDKErrorResult  # pylint: disable=E0611,E0401
```
Comment (Zeskbest) [inline lunavl/sdk/errors/exceptions.py:7]: E0401 does not exist
http://pylint-messages.wikidot.com/all-codes

**Review by Zeskbest — CHANGES_REQUESTED** [review]

File: lunavl/sdk/base.py (hunk 6 above)
Comment (matemax) [reply lunavl/sdk/base.py:106]: http://pylint.pycqa.org/en/latest/technical_reference/features.html?highlight=W0235#pylint-checkers-options-and-switches

File: lunavl/sdk/errors/exceptions.py (hunk 8 above)
Comment (matemax) [reply lunavl/sdk/errors/exceptions.py:7]: http://pylint.pycqa.org/en/latest/technical_reference/features.html?highlight=E0401#imports-checker-messages

**Pull request merged**